tensorflow_probability/python/experimental/distributions/laplace_approximation.py

```diff
@@ -0,0 +1,216 @@
+# Copyright 2021 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""A Laplace approximation over a JointDistribution object."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow.compat.v2 as tf
+
+from tensorflow_probability.python.bijectors import reshape
+from tensorflow_probability.python.bijectors import split
+from tensorflow_probability.python.bijectors import chain
+from tensorflow_probability.python.bijectors import joint_map
+from tensorflow_probability.python.distributions import transformed_distribution
+from tensorflow_probability.python.distributions import mvn_tril
+from tensorflow_probability.python.math import value_and_gradient
+from tensorflow_probability.python.optimizer import bfgs_utils
+from tensorflow_probability.python.optimizer import bfgs
+
+
+def _common_dtype(dtypes, dtype_hint=None):
```

Comment:
should be able to use `from tensorflow_probability.python.internal import dtype_util` and `dtype_util.common_dtype(dist._get_single_sample_distributions())`

Reply:
ah cool, was wondering how to use that function, done in 891de2793